compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -161,8 +161,8 @@ object DesugarEnums {
       val (tagMeth, scaffolding) = enumTagMeth(CaseKind.Object)
       val impl1 = cpy.Template(impl)(body =
         impl.body ++ List(tagMeth, toStringMeth) ++ registerCall)
-      val vdef = ValDef(name, TypeTree(), New(impl1)).withMods(mods | Final).withPos(pos)
-      flatTree(scaffolding ::: vdef :: Nil).withPos(pos.startPos)
+      val vdef = ValDef(name, TypeTree(), New(impl1)).withMods(mods | Final)
+      flatTree(scaffolding ::: vdef :: Nil).withPos(pos)
```

Comment:
The previous version called `withPos(pos.startPos)` on the whole tree, since children positions have to be contained inside their parents positions, this means that the whole tree including `vdef` got position `pos.startPos`.